File: src/main/java/rbq/wtf/lycoris/client/wrapper/wrappers/annotation/WrapClass.java
package rbq.wtf.lycoris.client.wrapper.wrappers.annotation;

import rbq.wtf.lycoris.client.wrapper.MapEnum;
import rbq.wtf.lycoris.client.wrapper.wrappers.annotation.repeat.WrapClasses;

import java.lang.annotation.*;

@Documented
@Target(ElementType.FIELD)
@Retention(RetentionPolicy.RUNTIME)
@Repeatable(WrapClasses.class)
public @interface WrapClass {
    String mcpName();
    MapEnum targetMap();
}

File: src/main/java/rbq/wtf/lycoris/client/wrapper/wrappers/annotation/WrapClass.kt
package rbq.wtf.lycoris.client.wrapper.wrappers.annotation

import rbq.wtf.lycoris.client.wrapper.MapEnum
import rbq.wtf.lycoris.client.wrapper.wrappers.annotation.repeat.WrapClasses

@MustBeDocumented
@Target(AnnotationTarget.FIELD)
@Retention(AnnotationRetention.RUNTIME)
@JvmRepeatable(
    WrapClasses::class
)
annotation class WrapClass(val mcpName: String, val targetMap: MapEnum)